Model/Component/Rewrites.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace CtiDigital\Configurator\Model\Component;
+
+use Magento\Framework\ObjectManagerInterface;
+use Magento\UrlRewrite\Model\UrlRewriteFactory;
+use Magento\UrlRewrite\Model\UrlPersistInterface;
+use CtiDigital\Configurator\Model\LoggingInterface;
+use CtiDigital\Configurator\Model\Exception\ComponentException;
+
+class Rewrites extends CsvComponentAbstract
+{
+    protected $alias = 'rewrites';
+    protected $name = 'rewrites';
+    protected $description = 'Component to create URL Store Rewrites';
+
+    /**
+     * @var UrlPersistInterface
+     */
+    protected $urlPersist;
+
+    /**
+     * @var UrlRewriteFactory
+     */
+    protected $urlRewriteFactory;
+
+    /**
+     * Rewrites constructor.
+     * @param LoggingInterface $log
+     * @param ObjectManagerInterface $objectManager
+     * @param UrlPersistInterface $urlPersist
+     * @param UrlRewriteFactory $urlRewriteFactory
+     */
+    public function __construct(
+        LoggingInterface $log,
+        ObjectManagerInterface $objectManager,
+        UrlPersistInterface $urlPersist,
+        UrlRewriteFactory $urlRewriteFactory
+    ) {
+        parent::__construct($log, $objectManager);
```

Comment:
If the implementation is not compatible with the parent it breaks Liskov Substitution principle. I think this will also generate a warning, which may cause problems in some cases.